core/src/edu/asu/c3simulator/widgets/NavigationPanel.java

```diff
@@ -0,0 +1,346 @@
+package edu.asu.c3simulator.widgets;
+
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+import com.badlogic.gdx.Game;
+import com.badlogic.gdx.Screen;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+
+/**
+ * This class creates a navigation panel by creating the buttons and linking them with
+ * their screens. It also supports having one level of sub-buttons to a parent button. A
+ * parent button will only show and hide sub-buttons when clicked. Only one of the parent
+ * buttons can show its sub-buttons at the same time. If a parent button is clicked, the
+ * already extended sub-buttons will collapse before extending the new set of sub-buttons.
+ * 
+ * @author Alyahya, Mohammed
+ */
+public class NavigationPanel extends Table
+{
+	private class NavigationButtonListener extends ClickListener
+	{
+		String screenName;
+		Screen targetedScreen;
+		
+		public NavigationButtonListener(String screenName, Screen targetedScreen)
+		{
+			this.screenName = screenName;
+			this.targetedScreen = targetedScreen;
+		}
+		
+		@Override
+		public void clicked(InputEvent event, float x, float y)
+		{
+			System.out.println(screenName);
+			game.setScreen(targetedScreen);
+		}
+	}
+	
+	private class CollapsibleButtonListener extends ClickListener
+	{
+		String screenName;
+		
+		public CollapsibleButtonListener(String screenName)
+		{
+			this.screenName = screenName;
+		}
+		
+		@Override
+		public void clicked(InputEvent event, float x, float y)
+		{
+			if (displayedCollapsibleButton.equalsIgnoreCase(screenName))
+				displayedCollapsibleButton = "";
+			else
+				displayedCollapsibleButton = screenName;
+			recreateNavigationPanel();
+		}
+	}
+	
+	private Game game;
+	private Skin skin;
+	private LinkedHashMap<String, Screen> buttons;
+	private LinkedHashMap<String, LinkedHashMap<String, Screen>> subButtons;
```

Comment:
I don't understand what you mean by using ButtonListeners. One of the reasons behind having the hash maps is to make it easier for the user of this class and decrease the  number of restrictions and very specific directions to use it as much as possible. Thus, now the user can add buttons any time and remove them and manipulate them based only on the button's name. Also, the way the code creates the navigation panel right now depends on them.

However, you are right the having all those hash maps is brutal. If there is time, finding a better way to implement this class might be a good idea.